imap/jmap_mail.c

```diff
@@ -12313,7 +12321,8 @@ static int _email_bulkupdate_plan_keywords(struct email_bulkupdate *bulk, ptrarr
     int i;
 
     /* Open seen.db, if required */
-    if (strcmp(bulk->req->accountid, bulk->req->userid)) {
+    if (bulk->from_accountid ||
```

Comment:
Maybe change `bulk->from_accountid` from `const char*` to `char *` which never is NULL? Would make it simpler to reason about the cost of malloc is small in comparison to the rest of the code anyway.